tests/test_config.py

```diff
@@ -1,23 +1,53 @@
 # pylint: disable=missing-module-docstring, missing-function-docstring, protected-access
 import unittest.mock
+from io import StringIO
+
+from test_self import _patch_streams
 
 import pylint.lint
 
 
-def check_configuration_file_reader(config_file):
-    """Initialize pylint with the given configuration file and check that
-    what we initialized the linter with what was expected.
+def run_with_config_file(config_file, args):
```

Comment:
```suggestion
def run_with_config_file(config_file;Union[Path, str], args: List[str]) -> Tuple[runner?, int, str]:
```
We do not have to maintain python 3.5 we can put typing in signature directly :)